packages/hmr-plugin/src/internal/hmr-lifecycle.ts

```diff
@@ -42,7 +42,7 @@ export class HmrLifecycle<T extends Partial<NgxsHmrLifeCycle<S>>, S> {
     if (typeof this.ngAppModule.hmrNgxsStoreBeforeOnDestroy === 'function') {
       state = this.ngAppModule.hmrNgxsStoreBeforeOnDestroy(ctx);
     } else {
-      state = ctx.getState();
+      state = ctx.getState() as Partial<S>;
```

Comment:
Leaving as this because I don't wanna do any changes in the HMR plugin, just making compiler happy...